### DISCUSSION TITLE
gNB with srsUE has a slow connection

### DISCUSSION THREAD

**Discussion by oguztudo** [srsran/srsRAN_Project#86] [General Help]
Hello everybody,

I managed to set up a connection using two USRP x310 devices, both running on separate PCs with Kubuntu 20.04 and UHD3.15.
The issue i am having right now is that the connection does not work as fast as expected. Here is a screenshot of the iperf3 where you can see that over the course of 10s traffic time with 10Mbit/s i get a quarter to a third as average throughput:

![iperf_srsue10M](https://github.com/srsran/srsRAN_Project/assets/129198919/b853c709-091a-41f4-b186-7fdf77a0fad9)

I have some Underruns on the UE side and losses on the UE side.
So the main question would be: Are there any settings (gNB, UE, USRP, UHD or Linux) that can improve my throughput or is this limitation purely on the hardware side?

Here are my Configuration files as well:

[gnb_x310.txt](https://github.com/srsran/srsRAN_Project/files/11625699/gnb_x310.txt)
[ue.txt](https://github.com/srsran/srsRAN_Project/files/11625717/ue.txt)

**Comment by pgawlowicz** [comment]
Looks like MCS in UL is low. This has to do with the low UL SNR. Could you try to increase the TX power of the UE?
